src/kvstore/kvstore_dist.h

```diff
@@ -45,12 +45,14 @@ namespace kvstore {
  * it's the server node's job to control the data consistency among all
  * workers. see details on \ref ServerHandle::Start
  */
+
+template <typename DType>
 class KVStoreDist : public KVStoreLocal {
```

Comment:
Is function Init also need to specify the dtype?

Reply:
Yes, pslite worker is defined in the init function.
We must let PSLite know which type data to transmit.

Reply:
But how to do it? I have checked your code in my mxnet, but it does not work. I have trained epoch 1 in single node with multiprecision and epoch 2 in 4 nodes. Epoch 1 worked well while epoch 2's acc decreased util 0.

Reply:
I have created KVstore with 'float16' and inited KVstroe_Server with dtype 'float16' .

Reply:
1. Try to train epoch 1 and 2 in same environment, such as 4 nodes.
2. How did you start training? I can make a test in my environment.

Reply:
Sorry, I have make a mistake in kvstore optimizer codes. I was tortured for several days with this error code. It can work well now.